src/sidebar/store/modules/toast-messages.js

```diff
@@ -0,0 +1,92 @@
+import * as util from '../util';
+
+/**
+ * A store module for managing a collection of toast messages. This module
+ * maintains state only; it's up to other layers to handle the management
+ * and interactions with these messages.
+ */
+
+function init() {
+  return {
+    messages: [],
+  };
+}
+
+const update = {
+  ADD_MESSAGE: function(state, action) {
+    return {
+      messages: state.messages.concat([{ ...action.message }]),
+    };
+  },
+
+  REMOVE_MESSAGE: function(state, action) {
+    const updatedMessages = state.messages.filter(
+      message => message.id !== action.id
+    );
+    return { messages: updatedMessages };
+  },
+
+  UPDATE_MESSAGE: function(state, action) {
+    const updatedMessages = state.messages.map(message => {
+      if (message.id && message.id === action.message.id) {
+        return { ...action.message };
+      }
+      return { ...message };
+    });
+    return { messages: updatedMessages };
+  },
+};
+
+const actions = util.actionTypes(update);
+
+function addMessage(message) {
+  return { type: actions.ADD_MESSAGE, message };
+}
+
+function removeMessage(id) {
+  return { type: actions.REMOVE_MESSAGE, id };
+}
+
+function updateMessage(message) {
+  return { type: actions.UPDATE_MESSAGE, message };
+}
+
+/** Selectors */
```

Comment:
I like this comment, can you add one for "Actions" above?

Reply:
I'm going to do this—I'm also going to add a discussion topic for our next frontend sync about the appropriate term for these functions, as they aren't truly actions, right?